front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx

```diff
@@ -56,6 +58,19 @@ export default function OSRDConfig({
 
   const shouldDisplayStdcmResult = selectedTrain !== undefined;
 
+  const enableApplyButton = () => {
+    if (
+      infra?.state === 'CACHED' &&
+      osrdconf.origin &&
+      osrdconf.destination &&
+      (osrdconf.originTime || osrdconf.destinationTime) &&
+      osrdconf.rollingStockID
+    ) {
+      return false;
+    }
+    return true;
+  };
```

Comment:
It's cleaner to return directly the boolean (rather than if () return false/true). Also you can use a const rather than a function (a useMemo would be even better !)
```suggestion
const disabledApplyButton = infra?.state !== 'CACHED' ||
    !osrdconf.origin ||
    !osrdconf.destination ||
    !(osrdconf.originTime || osrdconf.destinationTime) ||
    !osrdconf.rollingStockID;
```

Reply:
done